src/Components/Common/SearchByMultipleFields.tsx

```diff
@@ -0,0 +1,220 @@
+    (option: SearchOption) => {
+      setSelectedOption(option);
+      setSearchValue(option.value || "");
+      setOpen(false);
+      inputRef.current?.focus();
+      onSearch(option.key, option.value);
+    },
+    [onSearch],
+  );
+
+  const handleSearchChange = useCallback(
+    (value: string) => {
+      setSearchValue(value);
+      onSearch(selectedOption.key, value);
+    },
+    [selectedOption, onSearch],
+  );
+
+  const handleKeyDown = useCallback(
+    (e: React.KeyboardEvent<HTMLInputElement>) => {
+      console.log(e.key);
+      if (e.key === "Escape") {
+        setSearchValue("");
+        onSearch(selectedOption.key, "");
+      }
+      if (e.key === "/") {
+        e.preventDefault();
+        setOpen(true);
+      }
+    },
+    [selectedOption, onSearch],
+  );
+
+  const renderSearchInput = useMemo(() => {
+    const commonProps = {
+      ref: inputRef,
+      value: searchValue,
+      onChange: (e: any) =>
+        handleSearchChange(e.target ? e.target.value : e.value),
+      onKeyDown: handleKeyDown,
+      className: cn(
+        "flex-grow border-none shadow-none focus-visible:ring-0 h-10",
+        inputClassName,
+      ),
+    };
+
+    switch (selectedOption.type) {
+      case "phone":
+        return (
+          <PhoneNumberFormField
+            name={selectedOption.key}
+            placeholder={t(selectedOption.placeholder)}
+            types={["mobile", "landline"]}
+            {...commonProps}
+          />
+        );
+      default:
+        return (
+          <Input
+            type="text"
+            placeholder={t(selectedOption.placeholder)}
+            {...commonProps}
+          />
+        );
+    }
+  }, [
+    selectedOption,
+    searchValue,
+    handleSearchChange,
+    handleKeyDown,
+    t,
+    inputClassName,
+  ]);
+
+  return (
+    <div className={className}>
+      <div className="flex items-center rounded-t-lg border-x border-t border-gray-200 bg-white">
+        <Popover open={open} onOpenChange={setOpen}>
+          <PopoverTrigger asChild>
+            <Button
+              variant="ghost"
+              className="focus:ring-0"
+              size="sm"
+              onClick={() => setOpen(true)}
+            >
+              <CareIcon icon="l-search" className="mr-2 h-4 w-4" />/
+            </Button>
+          </PopoverTrigger>
+          <PopoverContent className="w-[200px] p-0">
+            <Command>
+              <CommandList>
+                <CommandGroup>
+                  {options.map((option) => (
+                    <CommandItem
+                      key={option.key}
+                      onSelect={() => handleOptionChange(option)}
+                    >
+                      <CareIcon icon="l-search" className="mr-2 h-4 w-4" />
+                      <span className="flex-1">{t(option.label)}</span>
+                      <kbd className="ml-auto text-xs text-gray-400">
+                        {option.label.charAt(0).toUpperCase()}
```

Comment:
The shortcut in the config should be used here. Also does the kbd work? 🤔 (I rember writing a catach all logic at top, that needs to rechecked)